components/nifosg/nifloader.cpp

```diff
@@ -1319,10 +1327,43 @@ namespace NifOsg
                 return nullptr;
             }
 
-            unsigned char* data = new unsigned char[pixelData->data.size()];
-            memcpy(data, pixelData->data.data(), pixelData->data.size());
+            const std::vector<unsigned char>& pixels = pixelData->data;
+            if (pixelData->fmt != Nif::NiPixelData::NIPXFMT_PAL8)
+            {
+                unsigned char* data = new unsigned char[pixels.size()];
+                std::copy(pixels.begin(), pixels.end(), data);
```

Comment:
Not sure what do you mean saying "more efficient", but `std::copy` may [use](https://godbolt.org/z/0fZGFY) `memmove` for implementation that is actually [slower](https://stackoverflow.com/questions/1201319/what-is-the-difference-between-memmove-and-memcpy) way to copy memory when compare with `memcpy`. Avoid to copy any memory would be prefered way if it is possible.

Reply:
[This stackoverflow question answers imply otherwise.](https://stackoverflow.com/questions/4707012/is-it-better-to-use-stdmemcpy-or-stdcopy-in-terms-to-performance).
Perhaps it's better to let the compiler decide for the coder how exactly to copy stuff and not assume the coder knows best.

Reply:
Just for reference: we use the `std::copy()` a lot more often in our codebase (excepts of `external` folder, IOCS and R&D use the `memcpy` internally) than `memcpy` since our general rule is to prefer C++ features over C ones.

But I'd recommend to do not touch the `memcpy` in this PR since it is a codestyle issue, not really related to paletted textures. IMO, it would be better to create a separate refactoring pull request to unify memory copying in OpenMW.

Reply:
Okay.